src/agent/src/rpc.rs

```diff
@@ -224,6 +224,14 @@ impl AgentService {
         // cannot predict everything from the caller.
         add_devices(&sl(), &req.devices, &mut oci, &self.sandbox).await?;
 
+        // In guest-kernel mode some devices need extra handling. Taking the
+        // GPU as an example the shim will inject CDI annotations that will
+        // be used by the kata-agent to do containerEdits according to the
+        // CDI spec coming from a registry that is created on the fly by UDEV
+        // or other entities for a specifc device.
+        handle_cdi_devices(&sl(), &req.devices, &mut oci, &self.sandbox).await?;
+        info!(sl(), "modified CDI container spec: {:?}", &oci);
```

Comment:
After handling cdi devices, it would be better if we can first verify whether the container is cdi container. If it is, we can then print the modified spec. Currently, the code prints 'modified CDI container spec' by default, regardless of whether it is a CDI container or not, which can cause confusion when checking the logs.

Reply:
Good catch. Will update.

Reply:
@ChengyuZhu6 Updated, only if we inject devices which means we have done containeredits we will print out the new spec.